pkg/cache/v3/snapshot.go

```diff
@@ -266,3 +269,30 @@ func (s *Snapshot) ConstructVersionMap() error {
 
 	return nil
 }
+
+// Respond to a delta watch with the provided snapshot value. If the response is nil, there has been no state change.
+func (s *Snapshot) respondDelta(ctx context.Context, request *DeltaRequest, value chan DeltaResponse, state stream.StreamState, log log.Logger) (*RawDeltaResponse, error) {
+	resources := &resourceContainer{
+		resourceMap:   s.GetResources(request.TypeUrl),
+		versionMap:    s.GetVersionMap(request.TypeUrl),
+		systemVersion: s.GetVersion(request.TypeUrl),
+	}
+	resp := createDeltaResponse(ctx, request, state, resources, log)
```

Comment:
I think now that we removed the logger from `createDeltaResponse` we need to remove it here too right?

Reply:
Could maybe do something like:
```go
resp := createDeltaResponse(ctx, request, state, resourceContainer{
    resourceMap:   s.GetResources(request.TypeUrl),
    versionMap:    s.GetVersionMap(request.TypeUrl),
    systemVersion: s.GetVersion(request.TypeUrl),
})

Reply:
done